### ISSUE TITLE
Allow user to specify whether TYPE_CHECKING should be true or not

### ISSUE THREAD

**Issue by lbenezriravin** [agronholm/sphinx-autodoc-typehints#58]
A recent commit 86d0834 sets `TYPE_CHECKING` to True unconditionally, which breaks a workaround I had been using for a bug in `typing`. Essentially, it works like this:

```
if TYPE_CHECKING:
    class MyType:
        """This is the actual class I need, but it falls over at runtime due to a bug"""
        ...
else:
    class MyType:
        """This is a dummy class"""
        ...
```

You can see the full thread here: https://github.com/python/typing/issues/561#issuecomment-398616670

Please make this variable user-configurable; I can't build my docs with this change. Thank you!

**Comment by agronholm** [comment]
I'm going to have to find out how to add configuration options to a Sphinx plugin.

**Comment by agronholm** [comment]
Fixed via 2afccd39f9e559f4cbd593461080cdfede7c2d50, v1.5.0 released.

**Comment by lbenezriravin** [comment]
Thank you for the fast turn around!
